include/h2o/balancer.h

```diff
@@ -41,16 +41,38 @@ typedef struct st_h2o_balancer_callbacks_t {
     h2o_balancer_destroyer destroy;
 } h2o_balancer_callbacks_t;
 
+typedef enum en_h2o_balancer_type_t {
+    H2O_BALANCER_TYPE_ROUND_ROBIN,
+    H2O_BALANCER_TYPE_LEAST_CONN,
+    H2O_BALANCER_TYPE_HASH
+} h2o_balancer_type_t;
```

Comment:
Isn't this like having a list of subclasses in the file that defines the superclass?

Reply:
This is configrator related so I think I could try to move it within the configrator file. Meanwhile for `h2o_balancer_hash_key_type_t` it could be not so obvious since the hashing key type is actually used by the callback which requesting the key for the specified request.